mdata/init.go

```diff
@@ -37,4 +45,52 @@ var (
 
 	// metric tank.gc_metric is the number of times the metrics GC is about to inspect a metric (series)
 	gcMetric = stats.NewCounter32("tank.gc_metric")
+
+	// set either via ConfigProcess or from the unit tests. other code should not touch
+	Schemas      persister.WhisperSchemas
+	Aggregations persister.WhisperAggregation
+
+	schemasFile = "/etc/metrictank/storage-schemas.conf"
+	aggFile     = "/etc/metrictank/storage-aggregation.conf"
 )
+
+func ConfigProcess() {
+	var err error
+
+	// === read storage-schemas.conf ===
+
+	// graphite behavior: abort on any config reading errors, but skip any rules that have problems.
+	// at the end, add a default schema of 7 days of minutely data.
+	// we are stricter and don't tolerate any errors, that seems in the user's best interest.
+
+	Schemas, err = persister.ReadWhisperSchemas(schemasFile)
+	if err != nil {
+		log.Fatalf("can't read schemas file %q: %s", schemasFile, err.Error())
+	}
+	Schemas = append(Schemas, persister.Schema{
+		Name:       "default",
+		Pattern:    regexp.MustCompile(""),
+		Retentions: whisper.Retentions([]whisper.Retention{whisper.NewRetentionMT(60, 3600*24*7, 120*60, 2, true)}),
```

Comment:
So these chunk spans, num chunks etc are all just hard coded? shouldn't that be configurable somehow?